src/main/java/net/sf/jabref/logic/net/Cookie.java

```diff
@@ -123,12 +112,16 @@ public boolean matches(URI uri) {
             return false;
         }
 
-        String path = uri.getPath();
-        if (path == null) {
-            path = "/";
+        String uriPath = uri.getPath();
+        if (uriPath == null) {
+            uriPath = "/";
         }
 
-        return path.startsWith(this.path);
+        return uriPath.startsWith(this.path);
+    }
+
+    public boolean equals(Cookie cookie) {
+        return ((domain.equals(cookie.domain)) && (name.equals(cookie.name)));
```

Comment:
I do not like this equals implementation as only two of the five attributes are used. What is more, when you override equals, you should also override hashcode.